From d0ea955a0c39c798ae6e3ff699659b90aa4be53a Mon Sep 17 00:00:00 2001
From: Thiraviyam Mariyappan <quic_tmariyap@quicinc.com>
Date: Tue, 14 Nov 2023 19:57:12 +0530
Subject: [PATCH] wifi: ath12k: Enable lock less tx netdevice flag

dev_fast_xmit spends more CPU cylces on spinlock before calling
ieee80211_subif_start_xmit_8023. Avoid the lock by enabling
NETIF_F_LLTX flag in netdev feature flag.

This lock less netdev tx provides 1GB throughput improvement
incase of 2link MLO.

Signed-off-by: P Praneesh <quic_ppranees@quicinc.com>
Signed-off-by: Thiraviyam Mariyappan <quic_tmariyap@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index 64f5661..6131156 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -16785,6 +16785,8 @@ static int ath12k_mac_hw_register(struct ath12k_hw *ah)
 		ieee80211_hw_set(hw, SUPPORT_FAST_XMIT);
 	}
 
+	hw->netdev_features |= NETIF_F_LLTX;
+
 	ret = ieee80211_register_hw(hw);
 	if (ret) {
 		ath12k_err(NULL, "ieee80211 registration failed: %d\n", ret);
-- 
2.17.1

